### ISSUE TITLE
Mesh CA proto is not shaded

### ISSUE THREAD

**Issue by ejona86** [grpc/grpc-java#7289]
I noticed `google.security.meshca.v1` in the build output, which is weird because it should be `com.google`; there was missing java_package and java_mulitple_files=true in the proto file. This was a tell-tale sign that the proto had not been used from Java before.

However, since it was in XDS, that is "okay" because all generated code in xds should be shaded and internal. However, looking at the configuration in xds/build.gradle it does not appear the google.security package is shaded.

It appears this leaking of generated code [made it into v1.31.0](https://github.com/grpc/grpc-java/commit/7047209ba35d35fc04f7425a21882ed15e228097). We should fix the shading and backport it to v1.31.x.

**Comment by sanjaypujare** [comment]
Since this file is imported from the Istio repo this is the plan:

- create a PR in the Istio repo to add the following to meshca.proto:
```
option java_multiple_files = true;
option java_outer_classname = "MeshCaProto";
option java_package = "com.google.security.meshca.v1";
```
- get the PR merged. Then update the import.sh script here to grab that version and have the proto file here synced up with that one.
- add shading in xds/build.gradle for the "com.google.security" package once the above is done.

For backporting to 1.31.0 I can just shade the existing package "google.security.meshca.v1" for now since the above steps could take some time.

Does that sound okay?

**Comment by ejona86** [comment]
> For backporting to 1.31.0 I can just shade the existing package "google.security.meshca.v1" for now since the above steps could take some time.

Let's do that first. We can then have a follow-up PR that syncs a nicer version of the .proto and renames the package being shaded.
